### PR TITLE
Fundable implementation

### PR DESCRIPTION
**Pull request by marcosio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,98 @@
...
+The unit tests use a JSON version of this standard, which can be seem below.
+
+```json
+{
+    "messageId": "Example Message ID",
+    "funds": [
+        {
+            "amount": 1.00,
+            "fundSourceId": "caaa2bd3-dc42-436a-b70b-d1d7dac23741",
+            "receiverInformation": "Example funds receiver information"
```
Comment (Ferparishuertas) [inline README.md]: I think this must me named fundingSubject, mainly you are funding your own wallet

File: contracts/Fundable.sol [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        FundableData storage fund = orderedFunds[operationId.toHash()];
+        require(fund.status == FundStatusCode.Ordered, "A fund can only be cancelled in status Ordered");
+        require(fund.walletToFund == msg.sender || fund.orderer == msg.sender, "Only the wallet who receives the fund can cancel");
+        fund.status = FundStatusCode.Cancelled;
+        emit FundCancelled(fund.orderer, operationId);
+        return true;
+    }
+
+    function processFund(string calldata operationId) external returns (bool) {
+        require(tokenOperator == msg.sender, "A fund can only be processed by the fund operator");
```
Comment (Ferparishuertas) [inline contracts/Fundable.sol]: I think we need to check if its  fundoperator. If yes, maybe a modifier can be implemented and used also in orderFrom

File: contracts/Fundable.sol [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        FundableData storage fund = orderedFunds[operationId.toHash()];
+        require(fund.status == FundStatusCode.Ordered, "A fund can only be put in process from status Ordered");
+        fund.status = FundStatusCode.InProcess;
+        emit FundInProcess(fund.orderer, operationId);
+        return true;
+    }
+
+    function executeFund(string calldata operationId) external returns (bool) {
+        require(tokenOperator == msg.sender, "A fund can only be executed by the fund operator");
```
Comment (Ferparishuertas) [inline contracts/Fundable.sol]: Same as above

File: contracts/Fundable.sol [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        string calldata operationId,
+        string calldata reason
+    ) external returns (bool)
+    {
+        FundableData storage fund = orderedFunds[operationId.toHash()];
+        require(
+            fund.status == FundStatusCode.Ordered || fund.status == FundStatusCode.InProcess,
+            "A fund can only be rejected if the status is ordered or in progress"
+        );
+        require(tokenOperator == msg.sender, "A fund can only be rejected by the token operator");
```
Comment (Ferparishuertas) [inline contracts/Fundable.sol]: Same as above

File: contracts/Fundable.sol (hunk 2 above)
Comment (daniel-iobuilders) [reply contracts/Fundable.sol]: In the Clearable and Payoutable we have used clearingAgent and payoutAgent. The advantage I see, is that we don't use operator for the contract owner and the operators that are set in authorizeFundOperator and can call orderFundFrom for somebody else.
So would fundAgent be okay for you as well?

Comment (Ferparishuertas) [reply contracts/Fundable.sol]: Makes sense, fundAgents and check if its one ok kind

File: README.md (hunk 1 above)
Comment (daniel-iobuilders) [reply README.md]: I've changed it now

File: contracts/Fundable.sol (hunk 2 above)
Comment (daniel-iobuilders) [reply contracts/Fundable.sol]: I've added a new role FundAgent and use this one

File: README.md (hunk 1 above)
Comment (Ferparishuertas) [reply README.md]: I meant the receivinginformation as fundingSubject  :) sorry

**Pull request merged**